src/std/misc/list.ss

```diff
@@ -265,3 +266,27 @@ package: std/misc
   (if (pair? lst)
     (take lst (1- (length lst)))
     lst))
+
+;; split the list lst into a list-of-lists using equals? and a symbol
+;; or a unary predicate procedure.
+;; (split '(1 2 a 3 4) 'a)                         => ((1 2) (3 4))
+;; (split '(a 1 2 a 3 4 a) 'a)                     => ((1 2) (3 4))
+;; (split '(1 2 "hi" 3 4) string?)                 => ((1 2) (3 4))
+;; (split '(1 2 a 3 4) (lambda (x) (equal? x 'a))) => ((1 2) (3 4))
+;; (split [] 'a)                                   => ()
+(def (split lst symbol-or-pred)
```

Comment:
I think that having a symbol or predicate is not very useful in general, it should just be a procedure.